user-interface/1004-ASSO-associate.md

```diff
@@ -1 +1,69 @@
+
+- **must** see the contract address for the governance token of the Vega network <a name="1004-ASSO-001" href="#1004-ASSO-001">1004-ASSO-001</a>
+
+...so I can participate in governance and staking.
+
+## Associate
+
+When looking to stake validators or participate in governance, I first need to associate governance tokens with a Vega wallet/key, I...
+
+- **must** [connect an Ethereum wallet/key](0004-EWAL-connect_ethereum_wallet.md) to see tokens it may have in wallet or attributed to it in the vesting contract <a name="1004-ASSO-002" href="#1004-ASSO-002">1004-ASSO-002</a>
+- **must** select a Vega key to associate to <a name="1004-ASSO-003" href="#1004-ASSO-003">1004-ASSO-003</a>
+  - **must** be able use a [connected Vega wallet](0002-WCON-connect_vega_wallet.md) as instead of manually inputting a public key <a name="1004-ASSO-004" href="#1004-ASSO-004">1004-ASSO-004</a>
+  - **should** be able to populate field with a string, so I can associate to a wallet I without connecting it
+- if the connected ethereum wallet has vesting tokens: **must** be able to select to associate from either the vesting contract or the wallet <a name="1004-ASSO-006" href="#1004-ASSO-006">1004-ASSO-006</a>
+- **must** see the number of un-associated tokens in the selected wallet/vesting contract <a name="1004-ASSO-007" href="#1004-ASSO-007">1004-ASSO-007</a>
+- **must** select the amount of tokens to associate <a name="1004-ASSO-008" href="#1004-ASSO-008">1004-ASSO-008</a>
+  - **must** be able to populate the input with the amount of un-associated tokens for the selected wallet/vesting contract <a name="1004-ASSO-009" href="#1004-ASSO-009">1004-ASSO-009</a>
+- **must** be warned if the amount being associated is greater than the amount available in the connected ethereum wallet <a name="1004-ASSO-010" href="#1004-ASSO-010">1004-ASSO-010</a>
+- **must** submit the association on [Ethereum transaction(s) inc ERC20 approval if required](0005-ETXN-submit_ethereum_transaction.md) <a name="1004-ASSO-011" href="#1004-ASSO-011">1004-ASSO-011</a>
+- **must** see feedback whether my association has been registered on Ethereum <a name="1004-ASSO-012" href="#1004-ASSO-012">1004-ASSO-012</a>
+- **must** see feedback that the association has been registered by Vega and that it can be used after the number of Ethereum block confirmations required (typically 50, check network param) <a name="1004-ASSO-013" href="#1004-ASSO-013">1004-ASSO-013</a>
+  - **should** be able to see a balance for the number of tokens associated that are ready for use <a name="1004-ASSO-014" href="#1004-ASSO-014">1004-ASSO-014</a>
+  - **should** be able to see a balance for the number of tokens for each pending association <a name="1004-ASSO-015" href="#1004-ASSO-015">1004-ASSO-015</a>
+- on completion: **should** be prompted to go on to [nominate](2001-STKE-staking.md) and/or participate in [Governance](1004-GOVE-governance_list.md)
+
+...so I can then use the Vega wallet to use my tokens.
+
```

Comment:
Not anything here about multiple wallets, either eth or vega side, and how that should behave. Do we want to specify that?

Reply:
003 and 004 talk about selecting a Vega key to associate too (there is also a should about inputing a string)
The connect to ETH wallet `0004-EWAL` does reference the idea that you can get multiple eth keys back. but I think this can assume you want to associate with the one connected eth wallet (because you have to).
or am I off? @dexturr